control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1741,6 +1748,13 @@ func (r *HostedControlPlaneReconciler) reconcileOpenShiftAPIServer(ctx context.C
 	}); err != nil {
 		return fmt.Errorf("failed to reconcile openshift apiserver deployment: %w", err)
 	}
+
+	// Ensure all replicas are updated and ready
+	// https://github.com/openshift/enhancements/blob/master/enhancements/update/eus-upgrades-mvp.md
+	if ready, err := util.IsDeploymentReady(ctx, r, deployment); !ready {
+		return fmt.Errorf("openshift apiserver deployment not ready: %w", err)
```

Comment:
Is returning an error here the best way to handle this?

It seems like we just need the reconcile look to break out early if we're waiting on something related to ordering before proceeding. This controller is watching this deployment. So any update to the deployment should already trigger a new reconcile. Otherwise we're going to spew a bunch of errors and exponentially back off during every install/update.

Reply:
Good point, I probably wouldn't classify this as an err. I can refactor a bit.

> This controller is watching this deployment. So any update to the deployment should already trigger a new reconcile.

If the deployment is applied and just simply waiting for the pods to become ready, I didn't think any updates to the status field would cause another reconciliation. Though I could be wrong.

We could potentially look into returning early and requeue.